### PR TITLE
add shadowsocks-rust

### PR DESCRIPTION
**Pull request by Dentrax**
* shadowsocks-rust-sslocal
> **Note**
> **Before:** `14.2MB, 15 packages, 0-CVE`
> **After:** `31.6MB, 10 packages, 0-CVE`

* shadowsocks-rust-ssserver
> **Note**
> **Before:** `12.5MB, 15 packages, 0-CVE`
> **After:** `29.9MB, 10 packages, 0-CVE`

### Image Size

- [ ] The Image is smaller in size than its common public counterpart.
- [X] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes: On both images, `/usr/lib/locale` is sized `18MB`.

image

### Image Vulnerabilities

- [X] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Image Tagging

- [ ] The image is _not_ tagged with version tags.
- [X] The image is tagged with :latest
- [ ] The image is not tagged with :latest (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [X] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [X] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [ ] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [X] A Helm chart was not provided.

Notes:

### Processor Architectures

- [X] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [X] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [X] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [X] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [X] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [X] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [X] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

Notes: ssserver requires root

### ENTRYPOINT

- [X] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [X] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [ ] Environment variables added.
- [X] Environment variables not added and not required.

### SIGTERM

- [X] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [X] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [X] A README file has been provided and it follows the README template.

### CODE REVIEW

File: images/shadowsocks-rust/main.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  source   = "./configs"
+  name     = each.key
+}
+
+module "latest" {
+  for_each = local.components
+  source   = "../../tflib/publisher"
+
+  name              = basename(path.module)
+  target_repository = var.target_repository
```
Comment (developer-guy) [inline images/shadowsocks-rust/main.tf:26]: we should add each component as suffix to the repository:

`"${var.target_repository}-${each.key}"`

File: images/shadowsocks-rust/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+* [View Image in Chainguard Academy](https://edu.chainguard.dev/chainguard/chainguard-images/reference/shadowsocks-rust/overview/)
+* [View Image Catalog](https://console.enforce.dev/images/catalog) for a full list of available tags.
+* [Contact Chainguard](https://www.chainguard.dev/chainguard-images) for enterprise support, SLAs, and access to older tags.*
+
+---
+<!--monopod:end-->
+
+<!--overview:start-->
+A Rust port of shadowsocks
```
Comment (mamccorm) [inline images/shadowsocks-rust/README.md:16]:
```suggestion
Shadowsocks-rust is a Rust implementation of the Shadowsocks protocol, aimed at ensuring secure and private internet access by encrypting connections and circumventing internet restrictions.
```

File: images/shadowsocks-rust/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name: shadowsocks-rust
+image: cgr.dev/chainguard/shadowsocks-rust
+logo: https://storage.googleapis.com/chainguard-academy/logos/shadowsocks-rust.svg
+endoflife: ""
+console_summary: ""
+short_description: A Rust port of shadowsocks
```
Comment (mamccorm) [inline images/shadowsocks-rust/metadata.yaml:6]:
```suggestion
Shadowsocks-rust is a Rust implementation of the Shadowsocks protocol, aimed at ensuring secure and private internet access by encrypting connections and circumventing internet restrictions.
```

File: images/shadowsocks-rust/README.md (hunk 2 above)
Comment (Dentrax) [reply images/shadowsocks-rust/README.md]: Thanks, done!

**Review by mamccorm — APPROVED** [review]

**Pull request merged**